server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/methods/MIMEMessageConverter.java

```diff
@@ -220,6 +215,46 @@ private Multipart createMultipart(CreationMessage newMessage, ImmutableList<Mess
         }
     }
 
+    private Multipart createMultipartWithAttachments(CreationMessage newMessage, ImmutableList<MessageAttachment> messageAttachments) throws IOException {
+        MultipartBuilder mixedMultipartBuilder = MultipartBuilder.create(MIXED_SUB_TYPE);
+        List<MessageAttachment> inlineAttachments = messageAttachments.stream()
+            .filter(MessageAttachment::isInline)
+            .collect(Guavate.toImmutableList());
+        List<MessageAttachment> besideAttachments = messageAttachments.stream()
+            .filter(att -> !att.isInline())
+            .collect(Guavate.toImmutableList());
+
+        if (!inlineAttachments.isEmpty()) {
+            Message innerMessage = Message.Builder.of()
+                .setBody(relatedMultipart(newMessage, inlineAttachments))
+                .build();
+
+            mixedMultipartBuilder.addBodyPart(innerMessage);
+        } else {
+            addBody(newMessage, mixedMultipartBuilder);
+        }
+
+        addAttachments(besideAttachments, mixedMultipartBuilder);
+
+        return mixedMultipartBuilder.build();
+    }
+
+    private Multipart relatedMultipart(CreationMessage newMessage, List<MessageAttachment> inlines) throws IOException {
+        MultipartBuilder relatedMultipart = MultipartBuilder.create(RELATED_SUB_TYPE);
+        addBody(newMessage, relatedMultipart);
+
+        return addAttachments(inlines, relatedMultipart)
+            .build();
+    }
+
+    private MultipartBuilder addAttachments(List<MessageAttachment> messageAttachments,
+                                            MultipartBuilder multipartBuilder) {
+        Consumer<MessageAttachment> addAttachment = addAttachment(multipartBuilder);
```

Comment:
This temporary variable do not add value.

Once you remove it the entire method stop making sens. Can we inline this?